src/workos.ts

```diff
@@ -214,6 +215,15 @@ export class WorkOS {
             requestID,
           });
         }
+        case 429: {
+          const retryAfter = headers.get('Retry-After');
+
+          throw new RateLimitExceededException(
+            data.message,
+            requestID,
+            retryAfter ? Number(retryAfter) : null,
+          );
+        }
```

Comment:
Is this a breaking change?

Existing users may be doing an `error instanceof GenericServerException` when attempting to handle rate limit errors, which would no longer work with the introduction of this new exception.

Reply:
Actually, one option would be to make `GenericServerException` the actual base-class that the other exceptions extend from. It can enforce that rough shape we want all of them to have, and would mean this is no longer a potential breaking-change.

Though I'd like to hear opinions before I make that change.

Reply:
Strongly prefer having `GenericServerException` be the base-class for inheritance to make this not be a breaking change. We want to generally shy away from releasing major versions too frequently. 

While someone out there might be relying on the `name`, I feel that that's a bit of an anti-pattern over checking `instanceof`. I'm comfortable with releasing this as a minor version instead of a major.

Reply:
I made the minimal change of only having the new exception inherit from `GenericServerException`, though a large change could be to have _all_ of the requests also inherit (and left a note about it): [4af4a79](https://github.com/workos/workos-node/pull/1033/commits/4af4a79e010b888466d32275650c1fb2c9faa874).